http/query_handler.go

```diff
@@ -98,20 +99,34 @@ func (h *FluxHandler) handleQuery(w http.ResponseWriter, r *http.Request) {
 		EncodeError(ctx, fmt.Errorf("unsupported dialect over HTTP %T", req.Dialect), w)
 		return
 	}
+
+	w.Header().Set("Trailer", QueryStatsTrailer)
 	hd.SetHeaders(w)
 
-	n, err := h.ProxyQueryService.Query(ctx, w, req)
+	cw := iocounter.Writer{Writer: w}
+	stats, err := h.ProxyQueryService.Query(ctx, &cw, req)
 	if err != nil {
-		if n == 0 {
+		if cw.Count() == 0 {
 			// Only record the error headers IFF nothing has been written to w.
 			EncodeError(ctx, err, w)
 			return
 		}
 		h.Logger.Info("Error writing response to client",
 			zap.String("handler", "flux"),
-			zap.Error(err),
-		)
+			zap.Error(err))
+		return
+	}
+
+	// Write statistics trailer
+	data, err := json.Marshal(stats)
+	if err != nil {
+		h.Logger.Info("Failed to encode statistics",
+			zap.String("handler", "flux"),
+			zap.Error(err))
+		return
 	}
+
+	w.Header().Set(QueryStatsTrailer, string(data))
```

Comment:
@goller You are correct, this should not be changed. The public API should not expose the trailer and should not be changing.